### ISSUE TITLE
python: refactor automatically generated bindings

### ISSUE THREAD

**Issue by djfritz** [sandia-minimega/minimega#328]
This needs a lot of discussion. We've done this twice already. 

@bjwrigh @jcrussell @michaelwisely

@bjwrigh please start this discussion.

**Comment by jcrussell** [comment]
Is there a problem with the bindings that we currently generate? The only issue that I know of is not wanting to vendorize all the python dependencies. Perhaps we should just to port the existing code to go? I like @devinc's approach otherwise.

**Comment by michaelwisely** [comment]
As far as the template goes, would `text/template` be a decent alternative to Jinja2? I've never used it before, but maybe it would remove the need for the `requirements.txt`.

**Comment by jcrussell** [comment]
Yup, I was looking at the template and I think it wouldn't be too difficult to port to `text/template`.

**Comment by devinc** [comment]
Yeah, probably makes sense for it to be in Go.

**Comment by bjwrigh** [comment]
Wild thought: spend this time making a RESTful web interface and let any
language use its own packages to use that API, alleviating the need to
manage bindings for every language someone uses

**Comment by djfritz** [comment]
The json speaking unix domain socket is supposed to be just that. Security is lended by the ACL on the file and we don't worry beyond that. A restful thing is just another layer of indirection - you add the restful interface and post a rest schema (or whatever they call it), and someone has to wrap that in their python. Better to just provide the existing json schema and wrap that.

Going back though, I think the part of this that needs refactoring is the interface given to the python end user. There should be a discussion on your side @bjwrigh regarding exactly what that should look like.

**Comment by djfritz** [comment]
Didn't mean to close this. Thanks github for putting that button next to and larger than the comment button.

**Comment by devinc** [comment]
...we already did that. You said we should wait to merge it because there
were big changes coming that made parts of it unnecessary.

**Comment by michaelwisely** [comment]
@devinc is [this](https://github.com/devinc/minimega/tree/67a70e41e6d4ec45fd001e88a6822185d80f0b3f/misc/python) what your'e referring to?

**Comment by devinc** [comment]
Nope, this summer we wrote a higher-level python API (based on minigiga
from @bjwrigh) that also provided things like groups/namespaces. It hasn't
been merged in since the scheduler is supposed to provide a lot of that at
the minimega level.
